Source/SwiftLintFramework/Rules/Idiomatic/ForceUnwrappingRule.swift

```diff
@@ -119,20 +121,18 @@ public struct ForceUnwrappingRule: OptInRule, ConfigurationProviderRule, Automat
                 .NSRangeToByteRange(start: secondRange.location, length: secondRange.length)
             else { return nil }
 
-        let kindsInFirstRange = syntaxMap.kinds(inByteRange: matchByteFirstRange)
-
         // check first captured range
-        // If not empty, first captured range is comment, string, keyword or typeidentifier.
+        // If not empty, first captured range is comment, string, typeidentifier or keyword that is not `self`.
         // We checks "not empty" because kinds may empty without filtering.
-        guard !kindsInFirstRange
-            .contains(where: ForceUnwrappingRule.excludingSyntaxKindsForFirstCapture.contains) else {
-                return nil
+        guard !isFirstRangeExcludedToken(byteRange: matchByteFirstRange, syntaxMap: syntaxMap, file: file) else {
+            return nil
         }
 
         let violationRange = NSRange(location: NSMaxRange(firstRange), length: 0)
+        let kindsInFirstRange = syntaxMap.kinds(inByteRange: matchByteFirstRange)
 
-        // if first captured range is identifier, generate violation
-        if kindsInFirstRange.contains(.identifier) {
+        // if first captured range is identifier or keyword (self), generate violation
+        if !Set(kindsInFirstRange).intersection([.identifier, .keyword]).isEmpty {
```

Comment:
Made a small improvement to this in a55178e68103fb51722f65bc5f921a8f99f08921

Reply:
Very cool, thanks!